matrix_sdk_base/src/store/sled_store/mod.rs

```diff
@@ -127,12 +129,45 @@ impl EncodeKey for (&str, &str, &str) {
     }
 }
 
+impl EncodeKey for (&str, &str, &str, &str) {
+    fn encode(&self) -> Vec<u8> {
+        [
+            self.0.as_bytes(),
+            &[ENCODE_SEPARATOR],
+            self.1.as_bytes(),
+            &[ENCODE_SEPARATOR],
+            self.2.as_bytes(),
+            &[ENCODE_SEPARATOR],
+            self.3.as_bytes(),
+            &[ENCODE_SEPARATOR],
+        ]
+        .concat()
+    }
+}
+
 impl EncodeKey for EventType {
     fn encode(&self) -> Vec<u8> {
         self.as_str().encode()
     }
 }
 
+/// Get the value at `position` in encoded `key`.
+///
+/// The key must have been encoded with the `EncodeKey` trait. `position`
+/// corresponds to the position in the tuple before the key was encoded. If it
+/// wasn't encoded in a tuple, use `0`.
+///
+/// Returns `None` if there is no key at `position`.
+pub fn decode_key_value(key: &[u8], position: usize) -> Option<String> {
+    let values: Vec<&[u8]> = key.split(|v| *v == ENCODE_SEPARATOR).collect();
+
+    if position >= values.len() {
```

Comment:
I think I prefer a `values.get(position).map(String::from_utf8_lossy)` type of thing here.